src/include/btmem.h

```diff
@@ -960,15 +946,71 @@ struct __wt_ref {
+     * operation is included in a checkpoint and then seen later, after a restart or via a
+     * checkpoint cursor.
+     *
+     * If the page must be instantiated for any reason: (1) WT_UPDATE structures are created for the
+     * page entries, (2) the transaction information from ft_info.del is copied to those WT_UPDATE
+     * structures (making them a match for the truncate operation), (3) the ft_info.del field is
+     * discarded, and (4) the WT_REF state switches to WT_REF_MEM.
+     *
+     * If the fast-truncate operation has not yet committed, additionally the ft_info.update field
+     * is created, which is an array of references to the WT_UPDATE structures, for subsequent
+     * transaction commit/abort. (The page can split, so there needs to be some way to find all of
+     * the update structures.)
+     *
+     * Doing anything other than testing if ft_info.del or ft_info.update is non-NULL (which
+     * eviction does) requires the WT_REF be locked.
+     *
+     * Because ft_info is a union it is important to always access the correct field. It is also
+     * vital to interpret the state correctly and consider all the possible cases.
+     *
+     * The union access should be ft_info.del if the state is WT_REF_DELETED (states 1 and 2 below),
+     * and should be ft_info.update if the state is WT_REF_MEM (states 5-6 below). Otherwise,
+     * neither field is valid and the pointer should always be NULL.
+     *
+     * These are the possible states:
+     *
+     * 1. The WT_REF state is WT_REF_DELETED and ft_info.del is NULL. This means the page is deleted
+     * and the deletion is globally visible. Any on-disk page has been or will be discarded.
+     *
+     * 2. The WT_REF state is WT_REF_DELETED and ft_info.del is not NULL. The page is on disk, and
+     * is deleted, but the deletion may not yet be globally visible (or visible to any given reader
+     * either). ft_info.del describes the delete operation. If it is necessary to read the page on
+     * behalf of a thread that cannot see the deletion, the page must be instantiated as described
+     * above.
+     *
+     * 3. The WT_REF state is WT_REF_DISK, and the parent page's address cell is a deleted-address
+     * cell. ft_info is not valid; ft_info.del should read as NULL. The page is on disk, and
+     * deleted; the deletion may not yet be globally visible. Because the time aggregate stored in
+     * the parent internal page includes the deletion time, tree walks will skip the page as
+     * appropriate without needing the fast-delete information. This state can only happen in
+     * readonly trees; it is a result of the page being read in and instantiated, but not marked
+     * dirty, then discarded by eviction. (In principle eviction should set the state back to
+     * WT_REF_DELETED in this case; however, this turns out to be awkward.) The only time this state
```

Comment:
```
(In principle eviction should set the state back to
     * WT_REF_DELETED in this case; however, this turns out to be awkward.)
```

sounds like a correctness issue, and it's not. Maybe "(In principle eviction could set the state back to
WT_REF_DELETED in this case, but it's simpler to do during the read path.)"

Reply:
Well, it was originally :-) and still would be if we didn't take account of it. I've rearranged the wording; see if you like the new text better. (In [42d20ae](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/42d20ae234f880c06e2cc2133a5c772b2c7414a6))